pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -1116,8 +1221,16 @@ func (lbaas *LbaasV2) EnsureLoadBalancerDeleted(clusterName string, service *v1.
 		return nil
 	}
 
-	if loadbalancer != nil && loadbalancer.VipPortID != "" {
+	// if this service has an annotation for floating network ID, use that instead of the default
+	floatingNetworkID := getStringFromServiceAnnotation(service, ServiceAnnotationLoadBalancerFloatingNetworkID, lbaas.opts.FloatingNetworkId, lbaas.opts.EnableAnnotationFloatingNetworkId)
+
+	if floatingNetworkID != "" && loadbalancer != nil && loadbalancer.VipPortID != "" {
 		portID := loadbalancer.VipPortID
+		port, err := getPortByIP(lbaas.network, loadbalancer.VipAddress)
```

Comment:
Bad code, please remove it, we can get portID by loadbalancer.VipPortID not getPortByIp().